### PR TITLE
[Base Screens - Auth] ForgotPasswordScreenBase

### PR DESCRIPTION
**Pull request by manojleaton**
Fixes #BLUI-5103 .

#### Changes proposed in this Pull Request:

- Added ForgotPasswordScreenBase component
-

#### Screenshots / Screen Recording (if applicable)
forgot-password

#### To Test:

- yarn start:example-android/ios
- Click on `ForgotPasswordScreenBase` button in navigation drawer

#### Any specific feedback you are looking for?

-

### CODE REVIEW

**Comment by ArshdeepSingh3-eaton** [comment]
testcase and documentation needs to be added

**Comment by ArshdeepSingh3-eaton** [comment]
testcase and documentation needs to be added

File: login-workflow/docs/screens/forgot-password.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+<AuthContextProvider {...props}>
+    <ForgotPasswordScreen />
+</AuthContextProvider>
+```
+
+## API
+
+| Prop Name           | Type                                         | Description                                                                                                                                              | Default                                       |
+| ------------------- | -------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------- |
+| emailLabel          | `string`                                     | Text to display as the label for the email text field.                                                                                                   | `t('bluiAuth:FORGOT_PASSWORD.EMAIL_ADDRESS')` |
```
Comment (ArshdeepSingh3-eaton) [inline login-workflow/docs/screens/forgot-password.md:22]: update default value to this format `t('bluiCommon.MESSAGE.OKAY') // "Okay"`

**Comment by ektaghag-eaton** [comment]
Email is invalid still I am able to click on next
Screenshot 2024-03-22 at 2 37 17 PM

**Comment by manojleaton** [comment]
> Email is invalid still I am able to click on next Screenshot 2024-03-22 at 2 37 17 PM
> 
> * not able to see the success screen after entering  valid email id.

It will not get render until we pass true value to the `showSuccessScreen` prop and it will work as you mention in the Full Screen component, which will be implemented in future.

**Comment by joebochill** [comment]
> It will not get render until we pass true value to the `showSuccessScreen` prop and it will work as you mention in the Full Screen component, which will be implemented in future.

The full screen will implement this behavior natively, yes. For the purposes of the demo and being able to test the screen, we should update our example file to have a local state variable to toggle the showSuccessScreen prop when the next button is clicked.

**Review by joebochill — COMMENTED** [review]
All of my comments I just left on ResetPassword also apply here.

**Review by surajeaton — APPROVED** [review]
Looks good.

**Pull request merged**